src/components/expression/expression-renderer.js

```diff
@@ -452,6 +515,99 @@ genotet.ExpressionRenderer.prototype.drawMatrixConditionLabels_ = function() {
 };
 
 /**
+ * Renders the expression heatmap gradient.
+ * @private
+ */
+genotet.ExpressionRenderer.prototype.drawHeatmapGradient_ = function() {
+  this.svgHeatmapGradient_.selectAll('*').remove();
+  var heatmapData = this.data.matrix;
+  var scaleMin = 0;
+  var scaleMax = 0;
+  if (this.data.options.autoScaleGradient) {
+    scaleMin = heatmapData.valueMin;
+    scaleMax = heatmapData.valueMax;
+  }
+  else {
+    scaleMin = heatmapData.allValueMin;
+    scaleMax = heatmapData.allValueMax;
+  }
+  scaleMin = scaleMin.toFixed(2);
+  scaleMax = scaleMax.toFixed(2);
+  var marginLeft = scaleMin.toString().length *
+    this.CONDITION_LABEL_HEIGHT_FACTOR_;
+  var marginRight = scaleMax.toString().length *
+    this.CONDITION_LABEL_HEIGHT_FACTOR_;
+
+  var transformLeft = 0;
+  var transformTop = this.heatmapHeight_;
+  if (this.data.options.showProfiles) {
+    transformTop += this.profileHeight_;
+  }
+  if (this.data.options.showGeneLabels) {
+    transformLeft += this.geneLabelWidth_ + this.LABEL_MARGIN_;
+  }
+  else if (this.data.options.showProfiles) {
+    transformLeft += this.DEFAULT_PROFILE_MARGIN;
+  }
+  if (this.data.options.showConditionLabels) {
+    transformTop += this.conditionLabelHeight_ +
+      this.LABEL_MARGIN_ + this.LABEL_DIFFERENCE_;
+  }
+  this.svgHeatmapGradient_
+    .attr(
+      'transform',
+      genotet.utils.getTransform([transformLeft, transformTop])
+    );
+  var gradientHeight = this.DEFAULT_HEATMAP_GRADIENT_HEIGHT -
+    this.HEATMAP_GRADIENT_MARGINS_.TOP -
+    this.HEATMAP_GRADIENT_MARGINS_.BOTTOM;
+  var gradientWidth = this.heatmapWidth_ - marginLeft - marginRight;
+  var gradientContent = this.svgHeatmapGradient_.append('defs')
+    .append('linearGradient')
+    .attr('id', 'gradient-content')
+    .attr('x1', '0%')
+    .attr('y1', '0%')
+    .attr('x2', '100%')
+    .attr('y2', '0%')
+    .attr('spreadMethod', 'pad');
+  gradientContent.append('stop')
+    .attr('offset', '0%')
+    .attr('stop-color', genotet.data.redYellowScale[0])
+    .attr('stop-opacity', 1);
+  gradientContent.append('stop')
+    .attr('offset', '50%')
+    .attr('stop-color', genotet.data.redYellowScale[1])
+    .attr('stop-opacity', 1);
+  gradientContent.append('stop')
+    .attr('offset', '100%')
+    .attr('stop-color', genotet.data.redYellowScale[2])
+    .attr('stop-opacity', 1);
+  this.svgHeatmapGradient_.append('rect')
+    .classed('gradient-rect', true)
+    .attr('transform', genotet.utils.getTransform([
```

Comment:
BTW if the gradient is always in a format like above, can't we make a static svg DOM and set its attr and visibility on demand? In this way we can avoid appending the same stuff repeatedly and reduce "hard coded html" in js source. Up to you though.